proof/refine/Retype_R.thy

```diff
@@ -12,88 +12,37 @@ theory Retype_R
 imports ArchVSpace_R
 begin
 
-context begin interpretation Arch . (*FIXME: arch-split*)
+arch_requalify_consts
+  no_gs_types
 
-definition
-  APIType_map2 :: "kernel_object + AARCH64_H.object_type \<Rightarrow> Structures_A.apiobject_type"
-where
- "APIType_map2 ty \<equiv> case ty of
-      Inr (APIObjectType ArchTypes_H.Untyped) \<Rightarrow> Structures_A.Untyped
-    | Inr (APIObjectType ArchTypes_H.TCBObject) \<Rightarrow> Structures_A.TCBObject
-    | Inr (APIObjectType ArchTypes_H.EndpointObject) \<Rightarrow> Structures_A.EndpointObject
-    | Inr (APIObjectType ArchTypes_H.NotificationObject) \<Rightarrow> Structures_A.NotificationObject
-    | Inr (APIObjectType ArchTypes_H.CapTableObject) \<Rightarrow> Structures_A.CapTableObject
-    | Inr LargePageObject \<Rightarrow> ArchObject LargePageObj
-    | Inr HugePageObject \<Rightarrow> ArchObject HugePageObj
-    | Inr PageTableObject \<Rightarrow> ArchObject PageTableObj
-    | Inr VSpaceObject \<Rightarrow> ArchObject VSpaceObj
-    | Inl (KOArch (KOASIDPool _)) \<Rightarrow> ArchObject ASIDPoolObj
-    | Inr VCPUObject \<Rightarrow> ArchObject AARCH64_A.VCPUObj
-    | _ \<Rightarrow> ArchObject SmallPageObj"
+lemmas [simp] = objBitsKO_less_word_bits
```

Comment:
Should we make this `[simp]` directly in the interface in `Bits_R`?

Reply:
done in next commit